package-res/resources/web/pentaho/type/Context.js

```diff
@@ -155,15 +240,63 @@ define([
     },
 
     /**
-     * Gets the instance constructor of a type.
+     * Gets, asynchronously, the **configured instance constructor** instance constructor of a type.
+     *
+     * For more information on the `typeRef` argument,
+     * please see [UTypeReference]{@link pentaho.type.spec.UTypeReference}.
+     *
+     * This method can be used even if a generic type metadata specification references
+     * non-standard types whose modules have not yet been loaded by the AMD module system.
+     *
+     * An error is thrown when:
```

Comment:
Another thing comes to my mind. Document that the method throws an error when an argument documented as being "required" is passed might be an anti-pattern. There's so much about a documented contract that we don't validate with errors. We just check the most obvious, causes of misuse, to aid the programmer in the programming effort and to find coding errors quicker.
These would be the errors that a compiler would catch, by static analysis — requiredness and type conformance — most of these we don't test because it would be expensive in performance and verbose. We could do this if we had a DEBUG variable setup along with dead-code elimination (we can actually do this with uglify).

What I do think should be documented as thrown errors are things like invalid arguments and operations.

Now, independently of the documentation, whatever we code, we need to test. So, even if we don't document the fact that no specifying a required argument actually throws an error, we need to test it :-)